src/cli/src/consumer/consume/mod.rs

```diff
@@ -74,9 +74,13 @@ pub struct ConsumeOpt {
     )]
     pub output: ConsumeOutputType,
 
-    /// Path to a WASM binary file
-    #[structopt(short, long)]
-    pub smart_stream: Option<PathBuf>,
+    /// Path to a SmartStream filter wasm file
+    #[structopt(long, group("smartstream"))]
+    pub filter: Option<PathBuf>,
```

Comment:
This will leads to proliferation of duplicate argument type.   For example, in the future, WASM could be loading not just from path but in the URL.  In that case, we have to change all these types.  Instead, we should decouple WASM from it's module type.  So instead, it's better if we have something like this:

```
pub struct SmartStreamConfig {
        pub module_source: Option<PathBuf>,
        pub module_type:  SmartStreamKind
```

Reply:
This will impact the user experience in the CLI,so @ajhunyady do you have any feedback on this? I think if we went with this approach, the CLI would look like:

```
fluvio consume <topic> --module-source=./smartstream.wasm --module-type=filter
fluvio consume <topic> --module-source=./smartstream.wasm --module-type=map
```

As opposed to what we had previously talked about:

```
fluvio consume <topic> --filter=./smartstream.wasm
fluvio consume <topic> --map=./smartstream.wasm
```